src/java/com/sarality/app/datastore/contact/ContactDataStore.java

```diff
@@ -203,8 +166,50 @@ private ContactDataBuilder extractContactData(ContactDataBuilder builder, long c
       nameBuilder.setMiddleName(processors.forString().extract(cursor, Column.DATA5));
       nameBuilder.setSuffix(processors.forString().extract(cursor, Column.DATA6));
       builder.setName(nameBuilder);
+    } else {
+      for (AppEnum app : AppEnum.values()) {
+        if (app.getMimeType().equals(mimeType)) {
+          builder.addAppContact(new AppContact(app, data, (long) processors.forInteger().extract(cursor, Column._ID)));
```

Comment:
If you expect a Long then use a processors.forLong() rather than processors.forInteger()